### PR TITLE
Add settleRebalanceAsync to rebalancing API.

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: src/api/RebalancingAPI.ts [hunk 1]
```diff
@@ -122,6 +122,27 @@ export class RebalancingAPI {
...
+  /**
+   * Settles rebalance after auction has been completed
+   *
+   * @param  rebalancingSetTokenAddress     Address of the Rebalancing Set
+   * @param  txOpts                         Transaction options
+   * @return                                Transaction hash
+   */
+  public async settleRebalanceAsync(
+    rebalancingSetTokenAddress: Address,
+    txOpts: TxData
```
Comment (justinkchen) [inline src/api/RebalancingAPI.ts]: trailing commas here and further on line 142

File: src/assertions/RebalancingAssertions.ts [hunk 2]
```diff
@@ -120,4 +134,23 @@ export class RebalancingAssertions {
       throw new Error(rebalancingErrors.INSUFFICIENT_TIME_PASSED(nextRebalanceFormattedDate));
     }
   }
+
+  /**
+   * Throws if given rebalancingSetToken is not in Rebalance state
```
Comment (justinkchen) [inline src/assertions/RebalancingAssertions.ts]: wrong comment

File: test/integration/api/RebalancingAPI.spec.ts [hunk 3]
```diff
@@ -441,4 +442,176 @@ describe('RebalancingAPI', () => {
       });
     });
   });
+
+  describe.only('settleRebalanceAsync', async () => {
```
Comment (justinkchen) [inline test/integration/api/RebalancingAPI.spec.ts]: remove the `.only`

File: src/api/RebalancingAPI.ts [hunk 4]
```diff
@@ -122,6 +122,27 @@ export class RebalancingAPI {
...
   }
 
+  /**
+   * Settles rebalance after auction has been completed
+   *
+   * @param  rebalancingSetTokenAddress     Address of the Rebalancing Set
+   * @param  txOpts                         Transaction options
+   * @return                                Transaction hash
+   */
+  public async settleRebalanceAsync(
```
[the remark this answers is not in the document]
Comment (bweick) [reply src/api/RebalancingAPI.ts]: Cool...I just assumed it was too long for the linter but I guess not.

File: src/assertions/RebalancingAssertions.ts [hunk 5]
```diff
@@ -64,6 +66,20 @@ export class RebalancingAssertions {
     }
   }
 
+  /**
+   * Throws if given rebalancingSetToken is not in Rebalance state
```
Comment (asoong) [inline src/assertions/RebalancingAssertions.ts]: YESS for these

**Comment by justinkchen** [comment]
LGTM

File: src/assertions/RebalancingAssertions.ts [hunk 6]
```diff
@@ -120,4 +136,149 @@ export class RebalancingAssertions {
       throw new Error(rebalancingErrors.INSUFFICIENT_TIME_PASSED(nextRebalanceFormattedDate));
     }
   }
+
+  /**
+   * Throws if not enough current sets rebalanced in auction
```
[the remark this answers is not in the document]
Comment (bweick) [reply src/assertions/RebalancingAssertions.ts]: It throws if someone calls settle and there are more sets remaining than the minimumBid amount. We can't be guaranteed the amount of currentSets collateralizing a rebalancingSetToken is divisible by the minimumBid amount...hence we have to allow for settlement to occur in that scenario where it's impossible for all currentSets to be rebalanced, that's why the threshold for settlement is not zero, but the minimumBid amount.

**Pull request merged**